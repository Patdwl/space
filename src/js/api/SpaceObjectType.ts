export enum SpaceObjectType {
  UNKNOWN = 0,
  PAYLOAD = 1,
  ROCKET_BODY = 2,
  DEBRIS = 3,
  SPECIAL = 4,
  RADAR_MEASUREMENT = 5,
  RADAR_TRACK = 6,
  RADAR_OBJECT = 7,
  BALLISTIC_MISSILE = 8,
  STAR = 9,
  INTERGOVERNMENTAL_ORGANIZATION = 10,
  SUBORBITAL_PAYLOAD_OPERATOR = 11,
  PAYLOAD_OWNER = 12,
  METEOROLOGICAL_ROCKET_LAUNCH_AGENCY_OR_MANUFACTURER = 13,
  PAYLOAD_MANUFACTURER = 14,
  LAUNCH_AGENCY = 15,
  LAUNCH_SITE = 16,
  LAUNCH_POSITION = 17,
  LAUNCH_FACILITY = 18,
  CONTROL_FACILITY = 19,
  GROUND_SENSOR_STATION = 20,
  OPTICAL = 21,
  MECHANICAL = 22,
  PHASED_ARRAY_RADAR = 23,
  OBSERVER = 24,
  BISTATIC_RADIO_TELESCOPE = 25,
  COUNTRY = 26,
  LAUNCH_VEHICLE_MANUFACTURER = 27,
  ENGINE_MANUFACTURER = 28,
}
