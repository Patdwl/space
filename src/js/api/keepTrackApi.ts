import { SatMath } from '../satMath/satMath';
import { KeepTrackApi, SatObject, SettingsManager } from './keepTrackTypes';
import { html } from './templateLiterals';

declare global {
  // eslint-disable-next-line no-unused-vars
  interface Window {
    settingsManager: SettingsManager;
    jQuery: unknown;
    $: unknown;
    gremlins: any;
    randomizer: any;
    keepTrackApi: KeepTrackApi;
    dataLayer: any; // For Google Tag Manager
    _numeric: any;
    satellite: SatMath;
    M: any;
  }
}

export const register = (params: { method: string; cbName: string; cb: any }) => {
  // If this is a valid callback
  if (typeof keepTrackApi.callbacks[params.method] !== 'undefined') {
    // Add the callback
    keepTrackApi.callbacks[params.method].push({ name: params.cbName, cb: params.cb });
  } else {
    throw new Error(`Invalid callback "${params.method}"!`);
  }
};
export const unregister = (params: { method: string; cbName: string }) => {
  // If this is a valid callback
  if (typeof keepTrackApi.callbacks[params.method] !== 'undefined') {
    for (let i = 0; i < keepTrackApi.callbacks[params.method].length; i++) {
      if (keepTrackApi.callbacks[params.method][i].name == params.cbName) {
        keepTrackApi.callbacks[params.method].splice(i, 1);
        return;
      }
    }
    // If we got this far, it means we couldn't find the callback
    throw new Error(`Callback "${params.cbName} not found"!`);
  } else {
    // Couldn't find the method
    throw new Error(`Invalid callback "${params.method}"!`);
  }
};
export const isThisJest = () => typeof process !== 'undefined';

export const keepTrackApi: KeepTrackApi = {
  html: html,
  register: register,
  unregister: unregister,
  callbacks: {
    selectSatData: [],
    updateSelectBox: [],
    onCruncherReady: [],
    onCruncherMessage: [],
    uiManagerInit: [],
    uiManagerOnReady: [],
    bottomMenuClick: [],
    hideSideMenus: [],
    nightToggle: [],
    orbitManagerInit: [],
    adviceReady: [],
    drawManagerLoadScene: [],
    drawOptionalScenery: [],
    updateLoop: [],
    rmbMenuActions: [],
    rightBtnMenuAdd: [],
    updateDateTime: [],
    uiManagerFinal: [],
    loadCatalog: [],
    resetSensor: [],
    setSensor: [],
    onDrawLoopComplete: [],
  },
  methods: {
    selectSatData: (sat: any, satId: number) => {
      keepTrackApi.programs.soundManager.play('whoosh');
      keepTrackApi.callbacks.selectSatData.forEach((cb: any) => cb.cb(sat, satId));
    },
    updateSelectBox: (sat: any) => {
      keepTrackApi.callbacks.updateSelectBox.forEach((cb: any) => cb.cb(sat));
    },
    onCruncherReady: () => {
      keepTrackApi.callbacks.onCruncherReady.forEach((cb: any) => cb.cb());
    },
    onCruncherMessage: () => {
      keepTrackApi.callbacks.onCruncherMessage.forEach((cb: any) => cb.cb());
    },
    uiManagerInit: () => {
      keepTrackApi.callbacks.uiManagerInit.forEach((cb: any) => cb.cb());
    },
    uiManagerOnReady: () => {
      keepTrackApi.callbacks.uiManagerOnReady.forEach((cb: any) => cb.cb());
    },
    bottomMenuClick: (iconName: string) => {
      keepTrackApi.programs.soundManager.play('genericBeep');
      keepTrackApi.callbacks.bottomMenuClick.forEach((cb: any) => cb.cb(iconName));
    },
    hideSideMenus: () => {
      keepTrackApi.callbacks.hideSideMenus.forEach((cb: any) => cb.cb());
    },
    nightToggle: (gl: any, nightTexture: any, texture: any) => {
      keepTrackApi.callbacks.nightToggle.forEach((cb: any) => cb.cb(gl, nightTexture, texture));
    },
    orbitManagerInit: () => {
      keepTrackApi.callbacks.orbitManagerInit.forEach((cb: any) => cb.cb());
    },
    adviceReady: () => {
      keepTrackApi.callbacks.adviceReady.forEach((cb: any) => cb.cb());
    },
    drawManagerLoadScene: () => {
      keepTrackApi.callbacks.drawManagerLoadScene.forEach((cb: any) => cb.cb());
    },
    drawOptionalScenery: () => {
      keepTrackApi.callbacks.drawOptionalScenery.forEach((cb: any) => cb.cb());
    },
    updateLoop: () => {
      keepTrackApi.callbacks.updateLoop.forEach((cb: any) => cb.cb());
    },
    rmbMenuActions: (menuName: string, satnum = -1) => {
      keepTrackApi.callbacks.rmbMenuActions.forEach((cb: any) => cb.cb(menuName, satnum));
    },
    rightBtnMenuAdd: () => {
      keepTrackApi.callbacks.rightBtnMenuAdd.forEach((cb: any) => cb.cb());
    },
    updateDateTime: (date: Date) => {
      keepTrackApi.callbacks.updateDateTime.forEach((cb: any) => cb.cb(date));
    },
    uiManagerFinal: () => {
      keepTrackApi.callbacks.uiManagerFinal.forEach((cb: any) => cb.cb());
    },
    loadCatalog: async (): Promise<SatObject[]> => keepTrackApi.callbacks.loadCatalog[0].cb(),
    resetSensor: () => {
      keepTrackApi.callbacks.resetSensor.forEach((cb: any) => cb.cb());
    },
    setSensor: (sensor: any, id: number) => {
      keepTrackApi.callbacks.setSensor.forEach((cb: any) => cb.cb(sensor, id));
    },
    onDrawLoopComplete: () => {
      keepTrackApi.callbacks.onDrawLoopComplete.forEach((cb: any) => cb.cb());
    },
  },
  programs: <any>{},
};

// First time we call this module we should make it available to the rest of the application
if (!window.keepTrackApi) window.keepTrackApi = keepTrackApi;
