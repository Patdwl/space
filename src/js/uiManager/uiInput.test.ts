import { keepTrackApiStubs } from '../api/apiMocks';
import { keepTrackApi } from '../api/keepTrackApi';
import { KeepTrackPrograms } from '../api/keepTrackTypes';
import * as uiInput from './uiInput';

keepTrackApi.programs = <KeepTrackPrograms>(<unknown>{ ...keepTrackApi.programs, ...keepTrackApiStubs.programs });
document.body.innerHTML = global.docBody;

// Let TypeScript know that settingsManager is a global variable
declare const settingsManager: any;

const keyEvt = (evt, param) => {
  const event = new window.KeyboardEvent(evt, param);
  window.dispatchEvent(event);
};

const mouseEvt = (evt, param, el) => {
  const event = new window.MouseEvent(evt, param);
  el = el || window;
  el.dispatchEvent(event);
};

describe('uiInput Base Tests', () => {
  test('UI Input Validation Test', () => {
    keyEvt('keydown', { which: '61', ctrlKey: true });

    settingsManager.disableWindowScroll = true;

    uiInput.init();
    window.dispatchEvent(new CustomEvent('scroll'));
    settingsManager.disableWindowScroll = false;
    settingsManager.disableNormalEvents = true;
    uiInput.init();
    settingsManager.disableUI = true;
    uiInput.init();
    settingsManager.disableUI = false;
    settingsManager.disableWindowScroll = false;
    settingsManager.disableNormalEvents = false;
    uiInput.init();
    keyEvt('keydown', { ctrlKey: true });
    keyEvt('keydown', { metaKey: true });
    keyEvt('keydown', {});
    keyEvt('keyup', { ctrlKey: false });
    keyEvt('keyup', { ctrlKey: true, metaKey: false });
    keyEvt('keyup', { ctrlKey: true, metaKey: true });
    keyEvt('keydown', { ctrlKey: true, which: 61 });
    keyEvt('keydown', { ctrlKey: true, which: 107 });
    keyEvt('keydown', { ctrlKey: true, which: 173 });
    keyEvt('keydown', { ctrlKey: true, which: 109 });
    keyEvt('keydown', { ctrlKey: true, which: 187 });
    keyEvt('keydown', { ctrlKey: true, which: 189 });

    mouseEvt('mousedown', { button: 1 }, null);
    mouseEvt('mouseup', { button: 1 }, null);
    mouseEvt('mousedown', { button: 2 }, null);
    mouseEvt('mouseup', { button: 2 }, null);
    keepTrackApi.programs.mainCamera.isShiftPressed = true;
    uiInput.init();
    mouseEvt('mousedown', { button: 1 }, null);
    mouseEvt('mouseup', { button: 1 }, null);
    mouseEvt('mousedown', { button: 2 }, null);
    mouseEvt('mouseup', { button: 2 }, null);
    settingsManager.disableCameraControls = true;
    mouseEvt('mousedown', { button: 1 }, null);
    mouseEvt('mouseup', { button: 1 }, null);

    const evObj = new Event('touchmove', { bubbles: true, cancelable: false });
    uiInput.canvasMouseDown(evObj);
    (<any>evObj).originalEvent = {
      touches: [
        { pageX: 100, pageY: 100 },
        { pageX: 100, pageY: 100 },
      ],
    };
    uiInput.canvasTouchStart(evObj);
    uiInput.canvasMouseUp(evObj);
    uiInput.openRmbMenu();
    uiInput.openRmbMenu(1);

    (<any>evObj).originalEvent = {
      touches: [{ pageX: 100, pageY: 100 }],
    };
    uiInput.canvasTouchStart(evObj);
  });
});

// @ponicode
describe('uiInput.init', () => {
  test('0', () => {
    let result: any = uiInput.init();
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('uiInput.unProject', () => {
  test('0', () => {
    let result: any = uiInput.unProject(350, 4);
    expect(result).toMatchSnapshot();
  });

  test('1', () => {
    let result: any = uiInput.unProject(50, 350);
    expect(result).toMatchSnapshot();
  });

  test('2', () => {
    let result: any = uiInput.unProject(1, 320);
    expect(result).toMatchSnapshot();
  });

  test('3', () => {
    let result: any = uiInput.unProject(1, 380);
    expect(result).toMatchSnapshot();
  });

  test('4', () => {
    let result: any = uiInput.unProject(350, 100);
    expect(result).toMatchSnapshot();
  });

  test('5', () => {
    let result: any = uiInput.unProject(NaN, NaN);
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('uiInput.getSatIdFromCoordAlt', () => {
  test('0', () => {
    let result: any = uiInput.getSatIdFromCoordAlt(410, 410);
    expect(result).toMatchSnapshot();
  });

  test('1', () => {
    let result: any = uiInput.getSatIdFromCoordAlt(70, 1);
    expect(result).toMatchSnapshot();
  });

  test('2', () => {
    let result: any = uiInput.getSatIdFromCoordAlt(100, 1);
    expect(result).toMatchSnapshot();
  });

  test('3', () => {
    let result: any = uiInput.getSatIdFromCoordAlt(400, 520);
    expect(result).toMatchSnapshot();
  });

  test('4', () => {
    let result: any = uiInput.getSatIdFromCoordAlt(1, 4);
    expect(result).toMatchSnapshot();
  });

  test('5', () => {
    let result: any = uiInput.getSatIdFromCoordAlt(NaN, NaN);
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('uiInput.getSatIdFromCoord', () => {
  test('0', () => {
    let result: any = uiInput.getSatIdFromCoord(350, 4);
    expect(result).toMatchSnapshot();
  });

  test('1', () => {
    let result: any = uiInput.getSatIdFromCoord(520, 90);
    expect(result).toMatchSnapshot();
  });

  test('2', () => {
    let result: any = uiInput.getSatIdFromCoord(320, 1);
    expect(result).toMatchSnapshot();
  });

  test('3', () => {
    let result: any = uiInput.getSatIdFromCoord(1, 50);
    expect(result).toMatchSnapshot();
  });

  test('4', () => {
    let result: any = uiInput.getSatIdFromCoord(30, 100);
    expect(result).toMatchSnapshot();
  });

  test('5', () => {
    let result: any = uiInput.getSatIdFromCoord(Infinity, Infinity);
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('uiInput.getEarthScreenPoint', () => {
  test('0', () => {
    let result: any = uiInput.getEarthScreenPoint(400, 100);
    expect(result).toMatchSnapshot();
  });

  test('1', () => {
    let result: any = uiInput.getEarthScreenPoint(100, 400);
    expect(result).toMatchSnapshot();
  });

  test('2', () => {
    let result: any = uiInput.getEarthScreenPoint(50, 550);
    expect(result).toMatchSnapshot();
  });

  test('3', () => {
    let result: any = uiInput.getEarthScreenPoint(350, 400);
    expect(result).toMatchSnapshot();
  });

  test('4', () => {
    let result: any = uiInput.getEarthScreenPoint(520, 50);
    expect(result).toMatchSnapshot();
  });

  test('5', () => {
    let result: any = uiInput.getEarthScreenPoint(NaN, NaN);
    expect(result).toMatchSnapshot();
  });
});

describe('uiInput.rmbMenuActions', () => {
  beforeAll(() => {
    window.M = {
      toast: jest.fn(),
    };
    keepTrackApi.programs.mainCamera.screenDragPoint = [0, 0, 0];
  });
  test('0', () => {
    let result: any = uiInput.rmbMenuActions(<any>{ target: { id: 'view-info-rmb' } });
    expect(result).toMatchSnapshot();
  });

  test('1', () => {
    let result: any = uiInput.rmbMenuActions(<any>{ target: { id: 'view-sat-info-rmb' } });
    expect(result).toMatchSnapshot();
  });

  test('2', () => {
    let result: any = uiInput.rmbMenuActions(<any>{ target: { id: 'view-sensor-info-rmb' } });
    expect(result).toMatchSnapshot();
  });

  test('3', () => {
    let result: any = uiInput.rmbMenuActions(<any>{ target: { id: 'view-related-sats-rmb' } });
    expect(result).toMatchSnapshot();
  });

  test('4', () => {
    let result: any = uiInput.rmbMenuActions(<any>{ target: { id: 'view-curdops-rmb' } });
    expect(result).toMatchSnapshot();
  });

  test('5', () => {
    let result: any = uiInput.rmbMenuActions(<any>{ target: { id: 'view-24dops-rmb' } });
    expect(result).toMatchSnapshot();
  });

  test('6', () => {
    let result: any = uiInput.rmbMenuActions(<any>{ target: { id: 'create-sensor-rmb' } });
    expect(result).toMatchSnapshot();
  });

  test('7', () => {
    let result: any = uiInput.rmbMenuActions(<any>{ target: { id: 'reset-camera-rmb' } });
    expect(result).toMatchSnapshot();
  });

  test('8', () => {
    let result: any = uiInput.rmbMenuActions(<any>{ target: { id: 'clear-lines-rmb' } });
    expect(result).toMatchSnapshot();
  });

  test('9', () => {
    let result: any = uiInput.rmbMenuActions(<any>{ target: { id: 'line-eci-axis-rmb' } });
    expect(result).toMatchSnapshot();
  });

  test('10', () => {
    let result: any = uiInput.rmbMenuActions(<any>{ target: { id: 'line-earth-sat-rmb' } });
    expect(result).toMatchSnapshot();
  });

  test('11', () => {
    let result: any = uiInput.rmbMenuActions(<any>{ target: { id: 'line-sensor-sat-rmb' } });
    expect(result).toMatchSnapshot();
  });

  test('12', () => {
    let result: any = uiInput.rmbMenuActions(<any>{ target: { id: 'line-sat-sat-rmb' } });
    expect(result).toMatchSnapshot();
  });

  test('13', () => {
    let result: any = uiInput.rmbMenuActions(<any>{ target: { id: 'line-sat-sun-rmb' } });
    expect(result).toMatchSnapshot();
  });

  test('15', () => {
    let result: any = uiInput.rmbMenuActions(<any>{ target: { id: 'create-observer-rmb' } });
    expect(result).toMatchSnapshot();
  });

  test('16', () => {
    let result: any = uiInput.rmbMenuActions(<any>{ target: { id: 'colors-default-rmb' } });
    expect(result).toMatchSnapshot();
  });

  test('17', () => {
    let result: any = uiInput.rmbMenuActions(<any>{ target: { id: 'colors-default-rmb' } });
    expect(result).toMatchSnapshot();
  });

  test('18', () => {
    let result: any = uiInput.rmbMenuActions(<any>{ target: { id: 'colors-sunlight-rmb' } });
    expect(result).toMatchSnapshot();
  });

  test('19', () => {
    let result: any = uiInput.rmbMenuActions(<any>{ target: { id: 'colors-country-rmb' } });
    expect(result).toMatchSnapshot();
  });

  test('20', () => {
    let result: any = uiInput.rmbMenuActions(<any>{ target: { id: 'colors-velocity-rmb' } });
    expect(result).toMatchSnapshot();
  });

  test('21', () => {
    let result: any = uiInput.rmbMenuActions(<any>{ target: { id: 'colors-ageOfElset-rmb' } });
    expect(result).toMatchSnapshot();
  });

  test('22', () => {
    let result: any = uiInput.rmbMenuActions(<any>{ target: { id: 'earth-blue-rmb' } });
    expect(result).toMatchSnapshot();
  });

  test('23', () => {
    let result: any = uiInput.rmbMenuActions(<any>{ target: { id: 'earth-nasa-rmb' } });
    expect(result).toMatchSnapshot();
  });

  test('24', () => {
    let result: any = uiInput.rmbMenuActions(<any>{ target: { id: 'earth-trusat-rmb' } });
    expect(result).toMatchSnapshot();
  });

  test('25', () => {
    let result: any = uiInput.rmbMenuActions(<any>{ target: { id: 'earth-low-rmb' } });
    expect(result).toMatchSnapshot();
  });

  test('26', () => {
    let result: any = uiInput.rmbMenuActions(<any>{ target: { id: 'earth-high-rmb' } });
    expect(result).toMatchSnapshot();
  });

  test('27', () => {
    let result: any = uiInput.rmbMenuActions(<any>{ target: { id: 'earth-high-no-clouds-rmb' } });
    expect(result).toMatchSnapshot();
  });

  test('28', () => {
    let result: any = uiInput.rmbMenuActions(<any>{ target: { id: 'earth-vec-rmb' } });
    expect(result).toMatchSnapshot();
  });

  test('29', () => {
    let result: any = uiInput.rmbMenuActions(<any>{ target: { id: 'earth-political-rmb' } });
    expect(result).toMatchSnapshot();
  });

  test('30', () => {
    let result: any = uiInput.rmbMenuActions(<any>{ target: { id: 'clear-screen-rmb' } });
    expect(result).toMatchSnapshot();
  });

  test('30', () => {
    let result: any = uiInput.rmbMenuActions(<any>{ target: { id: 'FAKE' } });
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('uiInput.hidePopUps', () => {
  test('0', () => {
    let result: any = uiInput.hidePopUps();
    expect(result).toMatchSnapshot();
  });
});

describe('uiInput.canvasMouseMove', () => {
  test('canvasMouseMove', () => {
    const canvasDom = document.createElement('canvas');
    keepTrackApi.programs.uiManager.uiInput.mouseMoveTimeout = -1;
    let result: any = uiInput.canvasMouseMove(<any>{ clientX: 100, clientY: 100 }, keepTrackApi.programs.mainCamera, canvasDom);
    expect(result).toMatchSnapshot();
  });
  test('canvasMouseMoveFire', () => {
    const canvasDom = {
      position: () => ({
        left: 100,
        top: 100,
      }),
    };
    keepTrackApi.programs.uiManager.uiInput.mouseMoveTimeout = -1;
    let result: any = uiInput.canvasMouseMoveFire(<any>{ clientX: 100, clientY: 100 }, keepTrackApi.programs.mainCamera, canvasDom);
    expect(result).toMatchSnapshot();
  });
});

describe('uiInput.canvasTouchMove', () => {
  test('0', () => {
    let result: any = uiInput.canvasTouchMove(<any>{ clientX: 100, clientY: 100 }, keepTrackApi.programs.mainCamera);
    expect(result).toMatchSnapshot();
  });
});

describe('uiInput.canvasTouchEnd', () => {
  test('0', () => {
    let result: any = uiInput.canvasTouchEnd(keepTrackApi.programs.mainCamera);
    expect(result).toMatchSnapshot();
  });
});
