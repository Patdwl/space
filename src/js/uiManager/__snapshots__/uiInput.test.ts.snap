// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uiInput.canvasMouseMove canvasMouseMove 1`] = `undefined`;

exports[`uiInput.canvasMouseMove canvasMouseMoveFire 1`] = `undefined`;

exports[`uiInput.canvasTouchEnd 0 1`] = `undefined`;

exports[`uiInput.canvasTouchMove 0 1`] = `undefined`;

exports[`uiInput.getEarthScreenPoint 0 1`] = `
Float32Array [
  NaN,
  NaN,
  NaN,
]
`;

exports[`uiInput.getEarthScreenPoint 1 1`] = `
Float32Array [
  NaN,
  NaN,
  NaN,
]
`;

exports[`uiInput.getEarthScreenPoint 2 1`] = `
Float32Array [
  NaN,
  NaN,
  NaN,
]
`;

exports[`uiInput.getEarthScreenPoint 3 1`] = `
Float32Array [
  NaN,
  NaN,
  NaN,
]
`;

exports[`uiInput.getEarthScreenPoint 4 1`] = `
Float32Array [
  NaN,
  NaN,
  NaN,
]
`;

exports[`uiInput.getEarthScreenPoint 5 1`] = `
Float32Array [
  NaN,
  NaN,
  NaN,
]
`;

exports[`uiInput.getSatIdFromCoord 0 1`] = `-1`;

exports[`uiInput.getSatIdFromCoord 1 1`] = `-1`;

exports[`uiInput.getSatIdFromCoord 2 1`] = `-1`;

exports[`uiInput.getSatIdFromCoord 3 1`] = `-1`;

exports[`uiInput.getSatIdFromCoord 4 1`] = `-1`;

exports[`uiInput.getSatIdFromCoord 5 1`] = `-1`;

exports[`uiInput.getSatIdFromCoordAlt 0 1`] = `undefined`;

exports[`uiInput.getSatIdFromCoordAlt 1 1`] = `undefined`;

exports[`uiInput.getSatIdFromCoordAlt 2 1`] = `undefined`;

exports[`uiInput.getSatIdFromCoordAlt 3 1`] = `undefined`;

exports[`uiInput.getSatIdFromCoordAlt 4 1`] = `undefined`;

exports[`uiInput.getSatIdFromCoordAlt 5 1`] = `undefined`;

exports[`uiInput.hidePopUps 0 1`] = `undefined`;

exports[`uiInput.init 0 1`] = `undefined`;

exports[`uiInput.rmbMenuActions 0 1`] = `undefined`;

exports[`uiInput.rmbMenuActions 1 1`] = `undefined`;

exports[`uiInput.rmbMenuActions 2 1`] = `undefined`;

exports[`uiInput.rmbMenuActions 3 1`] = `undefined`;

exports[`uiInput.rmbMenuActions 4 1`] = `undefined`;

exports[`uiInput.rmbMenuActions 5 1`] = `undefined`;

exports[`uiInput.rmbMenuActions 6 1`] = `undefined`;

exports[`uiInput.rmbMenuActions 7 1`] = `undefined`;

exports[`uiInput.rmbMenuActions 8 1`] = `undefined`;

exports[`uiInput.rmbMenuActions 9 1`] = `undefined`;

exports[`uiInput.rmbMenuActions 10 1`] = `undefined`;

exports[`uiInput.rmbMenuActions 11 1`] = `undefined`;

exports[`uiInput.rmbMenuActions 12 1`] = `undefined`;

exports[`uiInput.rmbMenuActions 13 1`] = `undefined`;

exports[`uiInput.rmbMenuActions 15 1`] = `undefined`;

exports[`uiInput.rmbMenuActions 16 1`] = `undefined`;

exports[`uiInput.rmbMenuActions 17 1`] = `undefined`;

exports[`uiInput.rmbMenuActions 18 1`] = `undefined`;

exports[`uiInput.rmbMenuActions 19 1`] = `undefined`;

exports[`uiInput.rmbMenuActions 20 1`] = `undefined`;

exports[`uiInput.rmbMenuActions 21 1`] = `undefined`;

exports[`uiInput.rmbMenuActions 22 1`] = `undefined`;

exports[`uiInput.rmbMenuActions 23 1`] = `undefined`;

exports[`uiInput.rmbMenuActions 24 1`] = `undefined`;

exports[`uiInput.rmbMenuActions 25 1`] = `undefined`;

exports[`uiInput.rmbMenuActions 26 1`] = `undefined`;

exports[`uiInput.rmbMenuActions 27 1`] = `undefined`;

exports[`uiInput.rmbMenuActions 28 1`] = `undefined`;

exports[`uiInput.rmbMenuActions 29 1`] = `undefined`;

exports[`uiInput.rmbMenuActions 30 1`] = `undefined`;

exports[`uiInput.rmbMenuActions 30 2`] = `undefined`;

exports[`uiInput.unProject 0 1`] = `
Array [
  NaN,
  NaN,
  NaN,
]
`;

exports[`uiInput.unProject 1 1`] = `
Array [
  NaN,
  NaN,
  NaN,
]
`;

exports[`uiInput.unProject 2 1`] = `
Array [
  NaN,
  NaN,
  NaN,
]
`;

exports[`uiInput.unProject 3 1`] = `
Array [
  NaN,
  NaN,
  NaN,
]
`;

exports[`uiInput.unProject 4 1`] = `
Array [
  NaN,
  NaN,
  NaN,
]
`;

exports[`uiInput.unProject 5 1`] = `
Array [
  NaN,
  NaN,
  NaN,
]
`;
